legacy/web/src/main/java/org/jboss/as/web/WebMigrateOperation.java

```diff
@@ -985,33 +982,36 @@ private void migrateConnector(OperationContext context, Map<PathAddress, ModelNo
                 if (scheme == null || scheme.equals("http")) {
                     newAddress = pathAddress(UndertowExtension.SUBSYSTEM_PATH, DEFAULT_SERVER_PATH, pathElement(Constants.HTTP_LISTENER, address.getLastElement().getValue()));
                     addConnector = createAddOperation(newAddress);
-                } else if (scheme.equals("https")) {
+                } else if (scheme.equals("https") && legacyAddOp == null ) {
```

Comment:
I think that if (scheme == null || scheme.equals("http") {...} else if (scheme.equals("https")) {final ModelNode legacyAddOp =...; if (legacyAddOpp == null) {...} else {...}} else {...} is cleaner and leaves no space for bugs. The way it is right now, you are saying that no matter what is the value of scheme (we only know it is not null and not equal to "http", if we have ssl we will go to http. Is that what you intended?

Reply:
You are right, only http and https values could be used but you can set there the value you like, added the changes. Thank you.